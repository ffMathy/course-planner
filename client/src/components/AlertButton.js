import React from "react";
import {
    Button,
    AlertDialog,
    AlertDialogContent,
    AlertDialogOverlay,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogHeader,
    IconButton,
} from "@chakra-ui/core";

const AlertButton = ({ itemType, itemName, action, navigateTo, confirmFn }) => {
    const [isOpen, setIsOpen] = React.useState();
    const onClose = () => setIsOpen(false);
    const cancelRef = React.useRef();

    return (
        <>
            <IconButton icon="delete" variantColor="red" size="lg" onClick={() => setIsOpen(true)} />
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                        {action} {itemType}
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        Are you sure you want to {action && action.toLowerCase()} {itemName}? You can't undo this action afterwards.
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose}>
                            Cancel
                        </Button>
                        <Button
                            variantColor="red"
                            onClick={() => {
                                onClose();
                                confirmFn();
                            }}
                            ml={3}
                        >
                            Delete
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default AlertButton;
