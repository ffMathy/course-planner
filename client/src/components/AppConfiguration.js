import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { theme, ThemeProvider, CSSReset } from "@chakra-ui/core";

/**
 * Wrapper Component to wrap all the different configurations
 * the main App Component uses.
 */

const AppConfiguration = ({ children }) => {
    return (
        <ThemeProvider theme={customTheme}>
            <CSSReset />
            <Router>{children}</Router>
        </ThemeProvider>
    );
};

// See: https://chakra-ui.com/getting-started
const customTheme = {
    ...theme,
};

export default AppConfiguration;
