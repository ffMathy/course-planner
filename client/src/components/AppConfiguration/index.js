import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { theme, ThemeProvider, CSSReset } from "@chakra-ui/core";

/**
 * Wrapper Component to wrap all the different configurations
 * the main App Component uses.
 */

const AppConfiguration = ({ children }) => {
    return (
        <ThemeProvider theme={theme}>
            <CSSReset />
            <Router>{children}</Router>
        </ThemeProvider>
    );
};

export default AppConfiguration;
