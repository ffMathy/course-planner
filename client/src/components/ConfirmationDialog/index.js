import React from "react";
import {
    Button,
    AlertDialog,
    AlertDialogOverlay,
    AlertDialogContent,
    AlertDialogHeader,
    AlertDialogBody,
    AlertDialogFooter,
} from "@chakra-ui/core";

const ConfirmationDialog = ({ isOpen, onClose, confirm, itemType, itemName, action, navigateTo }) => {
    const cancelRef = React.useRef();

    return (
        <>
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                        {action} {itemType}
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        Are you sure you want to {action && action.toLowerCase()} {itemName}? You can't undo this action afterwards.
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose}>
                            Cancel
                        </Button>
                        <Button variantColor="red" onClick={confirm} ml={3}>
                            {action}
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default ConfirmationDialog;
