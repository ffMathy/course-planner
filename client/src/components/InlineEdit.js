import React from "react";
import { Editable, EditableInput, EditablePreview, Box, Text } from "@chakra-ui/core";

const InlineEdit = ({ onSubmit, onCancel, onChange, value, title, placeholder }) => {
    return (
        <Editable
            value={value}
            onSubmit={onSubmit}
            onCancel={onCancel}
            onChange={onChange}
            marginTop="8px"
            textAlign="center"
            fontSize="2xl"
            display="flex"
            flexDirection="row"
            justifyContent="center"
            alignContent="center"
            placeholder={placeholder}
        >
            <Box>
                <Text> {title}: </Text>
            </Box>
            <Box>
                <EditablePreview ml={3} height={10} />
            </Box>
            <Box>
                <EditableInput ml={3} height={10} />
            </Box>
        </Editable>
    );
};

export default InlineEdit;
