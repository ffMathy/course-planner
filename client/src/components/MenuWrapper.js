import React from "react";
import { Flex } from "@chakra-ui/core";
import { OptionsMenu, ConfirmationDialog } from ".";

const MenuWrapper = ({ item, itemType, setOpenConfirmationDialog, openConfirmationDialog, confirm, hasEdit, onEdit }) => {
    return (
        <Flex>
            <OptionsMenu confirm={confirm} hasEdit={hasEdit} onEdit={onEdit} onDelete={() => setOpenConfirmationDialog(true)} />
            <ConfirmationDialog
                item={item}
                itemType={itemType}
                action={"Delete"}
                isOpen={openConfirmationDialog}
                onClose={() => setOpenConfirmationDialog(false)}
                confirm={confirm}
            />
        </Flex>
    );
};

export default MenuWrapper;
