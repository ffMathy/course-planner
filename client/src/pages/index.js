import HomePage from "./HomePage";
import CoursesPage from "./CoursesPage";
import ProgrammesPage from "./Programmes/ProgrammesPage";
import Students from "./Students";

export { HomePage, CoursesPage, ProgrammesPage, Students };
