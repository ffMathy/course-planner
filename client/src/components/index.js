import AppConfiguration from "./AppConfiguration";
import OutlineButton from "./OutlineButton";
import ProgrammeCard from "./ProgrammeCard";
import SearchBar from "./SearchBar";
import ProgrammeRequirementsItem from "./ProgrammeRequirementsItem";
import ConfirmationDialog from "./ConfirmationDialog";

export { AppConfiguration, OutlineButton, ProgrammeCard, SearchBar, ProgrammeRequirementsItem, ConfirmationDialog };
