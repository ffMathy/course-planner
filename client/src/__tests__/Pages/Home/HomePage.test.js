import React from "react";
import renderer from "react-test-renderer";
import { Home } from "../../../pages";
import { AppConfiguration } from "../../../components";

test("<Home />", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <Home />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
