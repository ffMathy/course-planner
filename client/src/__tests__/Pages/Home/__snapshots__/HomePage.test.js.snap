// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> 1`] = `
<div
  className="css-qxc4md"
>
  <div
    className="css-uxy78k"
  >
    <div
      className="css-1u9fuhx"
    >
      <button
        aria-controls="menu-1"
        aria-expanded={false}
        aria-haspopup="menu"
        className="css-wk9st5"
        id="menubutton-2"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="css-thzv76"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </button>
    </div>
    <div
      className="css-1lads1q"
    >
      <img
        alt="Homepage Image of Course Planner's screeshots"
        className="css-1phd9a0"
        height="400px"
        width="1000px"
      />
    </div>
  </div>
  <div
    className="css-gqkmby"
  >
    <div
      className="css-trwzgo"
    >
      <div
        className="css-gmuwbf"
      >
        <img
          className="css-0"
        />
      </div>
      <div
        className="css-1orkoap"
      >
        <p
          className="css-18zyayk"
        >
          Course Planner
        </p>
      </div>
      <div
        className="css-u6v8er"
      >
        <a
          href="/students"
          onClick={[Function]}
        >
          <button
            className="css-sw1q1f"
            type="button"
          >
            <p
              className="css-181lpyk"
            >
              Students
            </p>
          </button>
        </a>
        <a
          href="/courses"
          onClick={[Function]}
        >
          <button
            className="css-sw1q1f"
            type="button"
          >
            <p
              className="css-181lpyk"
            >
              Courses
            </p>
          </button>
        </a>
        <a
          href="/programmes"
          onClick={[Function]}
        >
          <button
            className="css-sw1q1f"
            type="button"
          >
            <p
              className="css-181lpyk"
            >
              Programmes
            </p>
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
